wallettool/src/main/java/org/bitcoinj/wallettool/WalletTool.java

```diff
@@ -1275,18 +1276,18 @@ private void printWallet(@Nullable AesKey aesKey) {
     }
 
     private void setCreationTime() {
-        long creationTime = getCreationTimeSeconds();
+        Optional<Instant> creationTime = getCreationTime();
         for (DeterministicKeyChain chain : wallet.getActiveKeyChains()) {
             DeterministicSeed seed = chain.getSeed();
             if (seed == null)
                 System.out.println("Active chain does not have a seed: " + chain);
             else
-                seed.setCreationTimeSeconds(creationTime);
+                creationTime.ifPresentOrElse(seed::setCreationTime, seed::clearCreationTime);
+
         }
-        if (creationTime > 0)
-            System.out.println("Setting creation time to: " + TimeUtils.dateTimeFormat(creationTime * 1000));
-        else
-            System.out.println("Clearing creation time.");
+        creationTime.ifPresentOrElse(
+                time -> System.out.println("Setting creation time to: " + TimeUtils.dateTimeFormat(time.toEpochMilli())),
+                () -> System.out.println("Clearing creation time."));
```

Comment:
```
String message = creationTime
                    .map(t -> "Setting creation time to: " + TimeUtils.dateTimeFormat(t))
                    .orElse("Clearing creation time.");
System.out.println(message);
```

Reply:
or:
```
System.out.println(creationTime.map(t -> "Setting creation time to: " + TimeUtils.dateTimeFormat(t))
                               .orElse("Clearing creation time."));
```

Reply:
I see, true functional style… just learned another bit, thanks!

I changed to your second example and pushed a commit on top.